### DISCUSSION TITLE
Are there thoughts of migrating to OpenSearch?

### DISCUSSION THREAD

**Discussion by lukasalexanderweber** [komoot/photon#728] [General]
As [Elasticsearch](https://en.wikipedia.org/wiki/Elasticsearch) is now proprietary software, I would like to know if you have plans to support the open source fork [OpenSearch](https://github.com/opensearch-project/OpenSearch)?

We saw your great presentation on the [FOSSGIS 2023](https://fossgis-konferenz.de/) and have good use cases for the software but we can not use Elasticsearch.

We might support the development if there is interest from your side in a change.

**Comment by otbutz** [comment]
This is mostly blocked by #325. Photon is currently stuck with an older ES release that still allows embedding. The biggest problem with migrating to a newer ES release or OpenSearch would be providing a good solution for testing, as outlined by @lonvia: https://github.com/komoot/photon/issues/325#issuecomment-812920427

**Reply by lukasalexanderweber** [reply]
thanks for the response! Nice to see Graphhopper devs around, do you use Photon for the search slot in graphhopper maps?

**Reply by otbutz** [reply]
I cannot really speak for the project as I am only an external contributor but GH allows to use photon as a geocoding provider: https://github.com/graphhopper/geocoder-converter#photon

**Comment by lonvia** [comment]
I currently don´t have the time to do this myself but would give support if you want to put development time into it.

The Lucene backend is still at bit away, so for a short-term solution I would favour a solution that supports 5.6 in embedded mode as well as newer versions of OpenSearch with external ES instances. It likely means that the code needs to be refactored to run with the old internal protocol as well as the HTTP client for newer versions. There are some experiments for running against the HTTP client [here](https://github.com/komoot/photon/compare/master...lonvia:photon:switch-to-http-client) but they were done before I refactored the code to work against Lucene and ES in parallel.

We don't use any fancy features of ES otherwise, so I don't think there are any major changes to the rest of the code necessary.

**Reply by lukasalexanderweber** [reply]
Thanks for the response! We'll need some time to discuss but I will get back to you later if this is an option for us. 

Thanks for the offer to support us and nice to hear that a switch to OpenSearch is not out of the question.
